source/features/restore-file.tsx

```diff
@@ -87,15 +88,48 @@ async function discardChanges(progress: (message: string) => void, originalFileN
 async function handleClick(event: DelegateEvent<MouseEvent, HTMLButtonElement>): Promise<void> {
 	const menuItem = event.delegateTarget;
 
-	const [originalFileName, newFileName = originalFileName] = menuItem
-		.closest('[data-path]')!
-		.querySelector('.Link--primary')!
-		.textContent
-		.split(' → ');
+	let originalFileName = '';
+	let newFileName = '';
+	// eslint-disable-next-line ts/no-restricted-types -- prompt also returns null
+	let commitTitle: undefined | null | string;
 
-	const commitTitle = prompt(`Are you sure you want to discard the changes to ${newFileName}? Enter the commit title`, `Discard changes to ${newFileName}`);
-	if (!commitTitle) {
-		return;
+	if (menuItem.tagName === 'A') {
+		const fileUrl = menuItem
+			.parentNode
+			?.parentNode
+			?.querySelector('li[data-variant="danger"] a')!
+			.href;
```

Comment:
1. Don't use `?.` but `!.` or else errors will just be silenced. We want errors to happen as early as possible
3. Use `parentElement` instead of `parentNode`
3. Use `.closest()` when traversing the DOM more than one `parent` away

Reply:
This is strange, but I can't seem to get `closest()` to find the element when selecting it via `'li[data-variant="danger"]'` (i.e. `$0.closest('li[data-variant="danger"]')`).

`$0` being the `menuItem` (i.e. the anchor tag in the `li` of our new btn). `closest` is supposed to walk up the tree as well, right? I also can't seem to get it to work adding another `.parentElement` before it, so that it only has to go up one level.

Alternatively, would you prefer this?

```js
const fileUrl = menuItem
	.parentElement!
	.parentElement!
	.querySelector('li[data-variant="danger"] a')!
	.href;
```


<img width="653" height="467" alt="CleanShot 2025-11-09 at 13 20 33" src="https://github.com/user-attachments/assets/d9c0577d-1b31-409f-8716-3542f82e4439" />

Reply:
If you can't get it to work, then the selector is wrong. `$0.closest(selector)` must work. You can't use `.rgh-restore-file`, how does that class get there in the first place?